test/app/sql_test.js

```diff
@@ -1,140 +1,143 @@
 const tmp = require('tmp');
-const assert = require('chai').assert;
+const {assert} = require('chai');
 
 const Sql = require('../../app/sql');
 
 describe('Sql', () => {
-    let tempRootDirectory = null;
-    let data = { 'conversationId': 'identifier',
-		 'expires_at': 1234,
-		 'hasAttachments': 1,
-		 'hasFileAttachments': 1,
-		 'hasVisualMediaAttachments': 1,
-		 'received_at': 123456,
-		 'schemaVersion': null,
-		 'sent_at': 1234567,
-		 'source': 'src',
-		 'sourceDevice': 'dev',
-		 'type': 0,
-		 'unread': 0,
-		 'expireTimer': 9876,
-		 'expirationStartTimestamp': 00000
-	       };
-    before(async () => {
-	tempRootDirectory = await tmp.dirSync().name;
-	let options = { 'configDir': tempRootDirectory, 'key': '010203098' };
-	await Sql.initialize(options);
-    });
-    
-    after(async () => {
-	await Sql.close();
-	await Sql.removeDB();
-    });
-    
-    it('saveMessage', async () => {
-	let count1 = await Sql.getMessageCount();
-	let id = await Sql.saveMessage(data);
-	let count2 = await Sql.getMessageCount();
-	assert.strictEqual(count2-count1, 1);
-	let data2 = Object.assign({}, data);
-	data2.id = 'modified';
-	let id2 = await Sql.saveMessage(data2, { 'forceSave': 0 });
-	let count3 = await Sql.getMessageCount();
-	assert.strictEqual(count3, count2);
-    });
+  let tempRootDirectory = null;
+  const data = {
+    conversationId: 'identifier',
+    expires_at: 1234,
+    hasAttachments: 1,
+    hasFileAttachments: 1,
+    hasVisualMediaAttachments: 1,
+    received_at: 123456,
+    schemaVersion: null,
+    sent_at: 1234567,
+    source: 'src',
+    sourceDevice: 'dev',
+    type: 0,
+    unread: 0,
+    expireTimer: 9876,
+    expirationStartTimestamp: 3248,
+  };
+  before(async () => {
+    tempRootDirectory = await tmp.dirSync().name;
+    const options = { configDir: tempRootDirectory, key: '010203098' };
+    await Sql.initialize(options);
+  });
 
-    it('removeMessage', async () => {
-	let count1 = await Sql.getMessageCount();
-	let id = await Sql.saveMessage(data);
-	await Sql.removeMessage(id);
-	let count2 = await Sql.getMessageCount();
-	assert.strictEqual(count2, count1);
-    });
+  after(async () => {
+    await Sql.close();
+    await Sql.removeDB();
+  });
 
-    it('getMessageById', async () => {
-	let id = await Sql.saveMessage(data);
-	let data2 = await Sql.getMessageById(id);
-	assert.strictEqual(data2.conversationId, data.conversationId);
-	assert.strictEqual(data2.expires_at, data.expires_at);
-	assert.strictEqual(data2.received_at, data.received_at);
-	assert.strictEqual(data2.type, data.type);
-	assert.strictEqual(data2.unread, data.unread);
-    });
+  it('saveMessage', async () => {
```

Comment:
Generally I like to use a `describe()` for the function (like `describe('#saveMessage`)` and then use `it()` calls inside that for the various scenarios that function needs to handle.

Generally, what this PR does is exercise the most basic of basic scenarios for each of these data access functions.

What is the goal here? We can get basic coverage of these functions by running the app.

If you'd like to cover tricky situations it's hard to cover when the app is behaving normally, that's interesting. If you'd like to comprehensively cover these functions so we're totally sure that they behave as expected, that's interesting. If you'd like to add tests that cover high throughput, high stress on the database, that would be interesting too.